avm/src/vm/run.rs

```diff
@@ -97,6 +99,11 @@ pub async fn run_file(fp: &str, delete_after_load: bool) {
   run(bytecode, HttpType::HTTP(HttpConfig { port: 8000, })).await;
 }
 
+async fn control_port(_req: Request<Body>) -> Result<Response<Body>, Infallible> {
+  // TODO: Add secret validation once the deploy service starts creating one for the cluster
+  Ok(Response::builder().status(200).body("ok".into()).unwrap())
```

Comment:
Let's make the health check smarter by checking if there are servers running in port `443` and/or port `8088`?

Reply:
Since this control port is only spun up in the situation where `443` is gonna be used, that part is easy, but how do we determine at the AVM level that an Anycloud app is running and that `8088` is also relevant?

Reply:
Currently, an alan app is an anycloud app to the daemon if the `Dockerfile` and `app.tar.gz` are present in `pwd` https://github.com/alantech/alan/blob/7245b348e9720b90921d45c6c3e623f174747502/avm/src/daemon/daemon.rs#L89

Reply:
Hmm... I can't use that particular logic since that's reading the file and then checking if that data was loaded correctly or not, but I can do something like that.

Reply:
As for testing the port, I could either just try to connect to `https://localhost` and `https://localhost:4142` and see what happens, or I can execute a shell command like this:

![Screenshot from 2021-03-16 14-36-17](https://user-images.githubusercontent.com/765551/111383071-fdedbe00-8664-11eb-813e-8c2c68969052.png)

Reply:
> Hmm... I can't use that particular logic since that's reading the file and then checking if that data was loaded correctly or not, but I can do something like that.

I know. We could generalize the "health" endpoint to provide a list of ports that we return status for all of them. Or we could just return the status for those 2 and transmit that in a similar manner to how we check for the files and transmit them.

Reply:
> As for testing the port, I could either just try to connect to `https://localhost` and `http://localhost:8088` and see what happens, or I can execute a shell command like this:
> 
> ![Screenshot from 2021-03-16 14-36-17](https://user-images.githubusercontent.com/765551/111383071-fdedbe00-8664-11eb-813e-8c2c68969052.png)

I was thinking a shell command? I'm not sure how we could connect to the port if we don't know the endpoint? You mean like a ping?

Reply:
After lots of discussion, decided re-using Rust's own TCP connection logic to test is cleaner and likely faster than shell scripts.

Here's a result of the new logic:

![Screenshot from 2021-03-16 15-30-21](https://user-images.githubusercontent.com/765551/111388567-c08d2e80-866c-11eb-952a-9b90e37e4cb9.png)